subsystem/kubeapi_test.go

```diff
@@ -834,6 +834,38 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		Expect(cluster.ImageGenerated).Should(Equal(true))
 	})
 
+	It("deploy infraEnv before clusterDeployment", func() {
+		infraEnvName := "infraenv"
+		secretRef := deployLocalObjectSecretIfNeeded(ctx, kubeClient)
+		clusterDeploymentSpec := getDefaultClusterDeploymentSNOSpec(secretRef)
+		infraEnvSpec := getDefaultInfraEnvSpec(secretRef, clusterDeploymentSpec)
+
+		deployInfraEnvCRD(ctx, kubeClient, infraEnvName, infraEnvSpec)
+		infraEnvKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      infraEnvName,
+		}
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		clusterKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+		checkClusterCondition(ctx, clusterKubeName, controllers.ClusterReadyForInstallationCondition, controllers.ClusterNotReadyReason)
+		cluster := getClusterFromDB(ctx, kubeClient, db, clusterKubeName, waitForReconcileTimeout)
+		configureLocalAgentClient(cluster.ID.String())
```

Comment:
Maybe `configureLocalAgentClient` should get `clusterKubeName` instead? I see this line 
```
getClusterFromDB(ctx, kubeClient, db, clusterKubeName, waitForReconcileTimeout)
```
happens a lot, and if it's only to get the ID. We can fetch the ID inside that function.

Reply:
Not related to this PR, but anyway in most of the places the cluster is used in other logic later in the code.